bobweb/bob/command_gpt.py

```diff
@@ -22,51 +27,85 @@ def __init__(self):
-        prompt = self.get_parameters(update.effective_message.text)
+        prompt = self.build_prompt_from_context()
 
         if not prompt:
             update.effective_message.reply_text("Anna jokin syöte komennon jälkeen. '[.!/]gpt [syöte]'", quote=False)
         else:
-            started_notification = update.effective_message.reply_text('Vastauksen generointi aloitettu. Tämä vie 30-60 sekuntia.', quote=False)
-            handle_response_generation_and_reply(update, prompt)
+            started_notification = update.effective_message.reply_text('Vastauksen generointi aloitettu.'
+                                                                       ' Tämä vie 30-60 sekuntia.', quote=False)
+
+            self.handle_response_generation_and_reply(update, prompt)
 
             # Delete notification message from the chat
             if context is not None:
-                context.bot.deleteMessage(chat_id=update.effective_message.chat_id, message_id=started_notification.message_id)
-
-
-def handle_response_generation_and_reply(update: Update, prompt: string) -> None:
-    try:
-        text_compilation = generate_and_format_result_text(prompt)
-        update.effective_message.reply_text(text_compilation)
-
-    except ResponseGenerationException as e:  # If exception was raised, reply its response_text
-        update.effective_message.reply_text(e.response_text, quote=True)
-
-
-def generate_and_format_result_text(prompt: string) -> string:
-    if os.getenv('OPENAI_API_KEY') is None:
-        logger.error('OPENAI_API_KEY is not set.')
-        raise EnvironmentError
-    openai.api_key = os.getenv('OPENAI_API_KEY')
-    completion = openai.ChatCompletion.create(
-        model='gpt-3.5-turbo',
-        messages=[
-                {'role': 'system', 'content': 'You are a helpful assistant.'},
-                {'role': 'user', 'content': prompt}
-            ]
-    )
-    content = completion.choices[0].message.content
-    cost = 'Cost of this query was: ${:f}'.format(completion.usage.total_tokens * 0.002 / 1000)
-    response = '{}\n\n{}'.format(content, cost)
-    return response
+                context.bot.deleteMessage(chat_id=update.effective_message.chat_id,
+                                          message_id=started_notification.message_id)
+
+    def build_prompt_from_context(self):
+        prompt = "Last " + str(self.conversation_context_length) + " messages from the chat:"
+        for update in self.conversation_context:
+            if update.effective_message.text is not None:
+                prompt += "\n\n" + str(database.get_telegram_user(update.effective_user.id)) + ": " \
+                          + update.effective_message.text
+        prompt += "'.\n\nThe last message starting with .gpt is addressed directly to you."
+        return prompt
+
+    def generate_and_format_result_text(self, prompt: string) -> string:
+        if os.getenv('OPENAI_API_KEY') is None:
```

Comment:
```suggestion
        if os.getenv('OPENAI_API_KEY') is None or os.getenv('OPENAI_API_KEY') == "":
```

Reply:
Näin ainakin muistaakseni kävi aiemmin että se oli olemassa (?) mutta siinä ei ollut arvoa